### PR TITLE
G Suite: Refine Add G Suite page and fix display issues

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request updates the `Add G Suite` page in order to refine a few things, and fix several display issues. More specifically, it:

* Fixes margin and alignment issues on small viewports
* Renames section and page titles to make the user experience more consistent with the previous page
* Fixes a wrong height of input fields with affixes (e.g. the one for the email address)
* Reduces the width of the button with the trash icon
* Temporarily hides this button on small viewports instead of displaying it in the middle of the form
* Changes the placeholder of the email input field to be more descriptive

These changes are preliminary work for #46204.

#### Large viewport

Before | After
------ | -----
screenshot | screenshot

#### Small viewport

Before | After
------ | -----
screenshot | screenshot

#### Testing instructions

1. Run `git checkout update/gsuite-add-user-page` and start your server, or open a [live branch](https://calypso.live/?branch=update/gsuite-add-user-page)
2. Open the [`Site Domains` page](http://calypso.localhost:3000/domains/manage)
3. Navigate to the `Email`page
4. Click the `Add New User` button
6. Assert that it looks like the screenshot above

### CODE REVIEW

**Comment by krymson24** [comment]
Looks awesome @stephanethomas !

Large Viewport
Screen Shot 2020-10-07 at 7 18 51 PM

Small Viewport
Screen Shot 2020-10-07 at 7 19 10 PM

File: client/components/gsuite/gsuite-new-user-list/style.scss [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
-	@include breakpoint-deprecated( '>800px' ) {
-		width: 75px;
+.gsuite-new-user-list__new-user {
+	fieldset {
+		margin-bottom: 0;
 	}
+}
+
+.gsuite-new-user-list__new-user-remove-user-button {
+	display: none; // TODO: Fix position of button which is displayed in the middle of input fields in small viewports
```
Comment (krymson24) [inline client/components/gsuite/gsuite-new-user-list/style.scss:8]: Is it customary to add a ticket for this TODO and include the issue in this comment as a reference?

**Review by krymson24 — APPROVED** [review]

File: client/my-sites/email/gsuite-add-users/index.jsx [hunk 2]
```diff
@@ -200,7 +200,7 @@ class GSuiteAddUsers extends React.Component {
 					return <QueryEmailForwards key={ domain.domain } domainName={ domain.domain } />;
 				} ) }
 
-				<SectionHeader label={ translate( 'Add G Suite' ) } />
+				<SectionHeader label={ translate( 'Add New User' ) } />
```
Comment (olaseni) [inline client/my-sites/email/gsuite-add-users/index.jsx:203]: Super minor and not a blocker.

A thought. As we can add multiple users with this interface, would it be more accurate to say **Add New Users**?

**Review by olaseni — APPROVED** [review]
Looks good!

File: client/my-sites/email/gsuite-add-users/index.jsx (hunk 2 above)
Comment (stephanethomas) [reply client/my-sites/email/gsuite-add-users/index.jsx:203]: That's a good point. I went for singular to be consistent with the label of the button that leads to this page. I will update that section header as well as this button to use plural.

File: client/components/gsuite/gsuite-new-user-list/style.scss (hunk 1 above)
Comment (stephanethomas) [reply client/components/gsuite/gsuite-new-user-list/style.scss:8]: That's a good idea, I'll update that comment. Thanks for the review @krymson24!

**Pull request merged**